src/solvers/strings/string_refinement.cpp

```diff
@@ -283,18 +283,10 @@ void string_refinementt::set_to(const exprt &expr, bool value)
 {
   PRECONDITION(expr.type().id() == ID_bool);
   PRECONDITION(equality_propagation);
-
-  if(expr.id() == ID_equal && value)
-  {
-    const equal_exprt &eq_expr = to_equal_expr(expr);
-    equations.push_back(eq_expr);
-  }
+  if(!value)
+    equations.push_back(not_exprt{expr});
```

Comment:
Is that to compensate for :racehorse: ?

Reply:
not really, it's more like the opposite: because we introduce things that may not be `equal_exprt` in `equations` they need to be filtered out in the previous commit